tests/qos/test_buffer.py

```diff
@@ -141,61 +170,135 @@ def check_pool_size(duthost, ingress_lossless_pool_oid, **kwargs):
         kwargs: The parameters based on which the expected pool size is calculated.
                 They are represeted in form of kwargs because different vendor can require different parameters
                 For Mellanox, it includes:
-                 - old / new pg size
-                 - old / new pg numbers
-                 - current pool size
-                 - the expected pool size is calculated as:
-                   current_pool_size + old_pg_num * old_pg_size - new_pg_num * new_pg_size
+                 - Old / new pg size
+                 - Old / new pg xoff (required only over subscribe ratio is defined)
+                 - Old / new pg numbers
+                 - Old_ratio / new_ratio (required only over subscribe ratio is defined)
+                 - Current pool size
+                 - Current shared headroom pool size (required only over subscribe ratio is defined)
+                 - The expected pool size is calculated as following:
+                    - Shared headroom pool disabled:
+                      current_pool_size + old_pg_num * old_pg_size - new_pg_num * new_pg_size
+                    - Shared headroom pool enabled by over subscribe ratio:
+                      current_pool_size + old_pg_num * old_pg_size - new_pg_num * new_pg_size
+                          + (old_pg_num * old_pg_xoff - new_pg_num * new_pg_xoff) * over_subscribe_ratio
     """
-    if duthost.facts['asic_type'] == 'mellanox':
-        old_headroom = int(kwargs["old_headroom"])
+    logging.debug("Kwargs {}".format(kwargs))
 
-        if "old_pg_number" in kwargs:
-            old_pg_number = int(kwargs["old_pg_number"])
+    if duthost.facts['asic_type'] == 'mellanox':
+        if kwargs.get("old_ratio") and kwargs.get("new_ratio"):
+            curr_pool_size = int(kwargs["pool_size"])
+            curr_shp_size = int(kwargs["shp_size"])
+            old_ratio = int(kwargs.get("old_ratio"))
+            new_ratio = int(kwargs.get("new_ratio"))
+            original_memory = curr_pool_size * DEFAULT_INGRESS_POOL_NUMBER + curr_shp_size
```

Comment:
Will this logic work when there are multiple ingress pools and they do not have the same size?

Reply:
No. But this is implemented on a per vendor basis and that is not the case for Mellanox.